packages/svelte/src/compiler/phases/3-transform/client/utils.js

```diff
@@ -418,15 +418,34 @@ export function serialize_set_binding(node, context, fallback, options) {
 			}
 		} else {
 			if (is_store) {
+				let left = /** @type {import('estree').Pattern} */ (visit(node.left));
+
+				// If we are assigning to a store property, we need to ensure we don't
+				// capture the read for the store as part of the member expression to
+				// keep consistency with how store $ shorthand reads work in Svelte 4.
+				if (
+					node.left.type === 'MemberExpression' &&
+					left.type === 'MemberExpression' &&
+					node.left.object.type === 'Identifier' &&
+					!node.left.computed
+				) {
```

Comment:
I don't think that we need any of these checks. The checks above already ensure that this is a store and that it's an object mutation. This is also buggy right now, if you do `$foo.bar.baz = newValue` this check wouldn't apply anymore.
I think the right logic would be to traverse to the call expression that gets the store that is now part of the `left` AST and wrap that one with `$.untrack`

Reply:
Pushed a more thorough fix.